common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -198,6 +198,7 @@ protected void serviceInit(Configuration conf) throws Exception {
       // register metrics on the block handler into the Node Manager's metrics system.
       blockHandler.getAllMetrics().getMetrics().put("numRegisteredConnections",
```

Comment:
If we put all metrics at line 201, do we need to have this line?

Reply:
Line 201 only adds all metrics from `NettyMemoryMetrics`. This line adds `numRegisteredConnections` from `TransportContext` which is actually counting the number of registrations as explained above.